### ISSUE TITLE
better protection of user home dir in nix-shell

### ISSUE THREAD

**Issue by wamserma** [Mic92/nixpkgs-review#126]
People often run reviews on their normal desktops. For some packages it even makes sense to execute the results under the regular user account for testing. However, this can modify configs in a non-backward compatible way.

For example [this](https://github.com/NixOS/nixpkgs/pull/96895#issuecomment-689323223) locked me out of my DE and prevented Firefox from launching. Debugging from the segfaults would have taken ages and a quick recovery was only possible as I luckily had seen the entry on the font cache issue minutes before.

Would it be reasonable to mount an Overlay FIlesystem for `$HOME` and point `HOME` and `XDG_*_HOME` there (in the resulting nix-shell)?

**Comment by Mic92** [comment]
Might be an option however overlayfs is not supported on all filesystems for example zfs is not supported but popular within the nixpkgs maintainer group.

**Comment by Mic92** [comment]
One alternative would be to use a different `$HOME` something like:

```
export HOME=$(mktemp -d)
```

Also has the advantage that state in `$HOME` does not affect testing.

**Comment by Mic92** [comment]
Something like this might even provide more protection: https://github.com/Mic92/nixos-shell however x11 support is a bit trickier.

**Comment by wamserma** [comment]
> One alternative would be to use a different `$HOME` something like:
> 
> ```
> export HOME=$(mktemp -d)
> ```
> 
> Also has the advantage that state in `$HOME` does not affect testing.

Yes, that would be reasonable. For testing one can still copy the relevant parts of the real home into the temp-$HOME.
